### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**

### CODE REVIEW

File: .github/workflows/github_actions.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      matrix:
+        os: [ ubuntu-18.04, macos-10.15 ]
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          unset SYSTEM
+          set -e pipefail
+          open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
+          sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -allowUntrusted -target /
```
Comment (ihnorton) [inline .github/workflows/github_actions.yml:24]: Is this still necessary? I removed it in tiledb-py recently and everything still worked.

File: .github/workflows/github_actions.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        shell: bash
+        name: Install_system_headers_mac_only
+        if: matrix.os == 'macOS-10.14'
+
+      - run: |
+          unset SYSTEM
```
Comment (ihnorton) [inline .github/workflows/github_actions.yml:30]: Is this necessary? It was an issue w/ AZP, but I don't know if GHA does this.

File: .github/workflows/github_actions.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        name: checkStyle
+
+      - run: |
+          unset SYSTEM
+          export JAVA_HOME=$JAVA_HOME_11_X64
+          mvn clean package -DskipTests
+        shell: bash
+        name: checkFormat
+
+      - run: |
```
Comment (ihnorton) [inline .github/workflows/github_actions.yml:43]: These could all be combined into a single `run: |` block, but make sure to add `set -eu` at the top so an individual failure will fail the step (assuming GHA has the same default as AZP).

**Review by ihnorton — APPROVED** [review]
Minor suggestions, otherwise LGTM.

File: .github/workflows/github_actions.yml (hunk 1 above)
Comment (DimitrisStaratzis) [reply .github/workflows/github_actions.yml]: It is not! I removed it.

File: .github/workflows/github_actions.yml (hunk 2 above)
Comment (DimitrisStaratzis) [reply .github/workflows/github_actions.yml:22]: Do you refer to the same thing as above?

File: .github/workflows/github_actions.yml (hunk 3 above)
Comment (DimitrisStaratzis) [reply .github/workflows/github_actions.yml:35]: Yes, but I chose to have the steps separately for better clarity when running the CI. Let me know what you think.

File: .github/workflows/github_actions.yml (hunk 2 above)
Comment (ihnorton) [reply .github/workflows/github_actions.yml:22]: Yes, nevermind

File: .github/workflows/github_actions.yml (hunk 3 above)
Comment (ihnorton) [reply .github/workflows/github_actions.yml:35]: Ok no problem

**Review by ihnorton — APPROVED** [review]

**Review by ihnorton — APPROVED** [review]
Please remove the `unset SYSTEM` lines, then LGTM.

**Pull request merged**